security/pkg/caclient/client.go

```diff
@@ -72,7 +67,7 @@ func NewCAClient(pltfmc platform.Client, ptclc grpc.CAGrpcClient, cAAddr string,
 
 // RetrieveNewKeyCert sends the CSR to Istio CA with automatic retries. When successful, it returns the generated key
 // and cert, otherwise, it returns error. This is a blocking function.
-func (c *cAClientImpl) RetrieveNewKeyCert() (newCert []byte, certChain []byte, privateKey []byte, err error) {
+func (c *CAClient) RetrieveNewKeyCert() (newCert []byte, certChain []byte, privateKey []byte, err error) {
```

Comment:
I think we need to modify the function signature to match Retrieve() (newCert, certChain, privateKey []byte, err error) to be able to use the interface KeyCertRetriever.

Reply:
Oh... Forgot this, fixed, thx!